experiment/secret-sync-controller/secret-sync-config.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	// b64 "encoding/base64"
+	// "bytes"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"strconv"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretSyncConfig struct {
+	Specs []SecretSyncSpec `yaml:"specs"`
+}
+
+type SecretSyncSpec struct {
+	// the target can be either a K8s secret or a SecretManager secret
+	Source      Target `yaml:"source"`
+	Destination Target `yaml:"destination"`
+}
+
+type Target struct {
+	// assert that one of the two should be `nil`
+	Kubernetes    *KubernetesSpec    `yaml:"kubernetes,omitempty"`
+	SecretManager *SecretManagerSpec `yaml:"secretManager,omitempty"`
+}
+
+type KubernetesSpec struct {
+	Namespace string   `yaml:"namespace"`
+	Secret    string   `yaml:"secret,omitempty"`
+	DenyList  []string `yaml:"denyList,omitempty"`
+}
+
+type SecretManagerSpec struct {
+	Project  string   `yaml:"project"`
+	Secret   string   `yaml:"secret,omitempty"`
+	DenyList []string `yaml:"denyList,omitempty"`
+}
+
+func (target Target) GetLatestSecretVersion(k8sClientset *kubernetes.Clientset, secretManagerCtx context.Context, secretManagerClient *secretmanager.Client) (int, map[string][]byte) {
```

Comment:
As we discussed this morning, prefer instead to return a single result and an error.  Returning an untyped tuple feels like python, no go.  So this case, something like
```suggestion
type struct Secret {
  Version string // I'm using string instead of int, see comment in LatestSecretVersion
  Data map[string][]byte
}

func (target Target) GetLatestSecretVersion(k8sClientset *kubernetes.Clientset, secretManagerCtx context.Context, secretManagerClient *secretmanager.Client) (Secret, error) {
```